jenkins/python-setup.sh

```diff
@@ -1,23 +1,22 @@
-_prefix="${1}"
+#!/bin/bash
 
-if [[ -z "${_prefix}" ]]; then
-    printf -- "Missing prefix argument!" >&2
-    exit 1
-fi
+# This script is run by jenkins/run-unittests to set up the Python environment
+# for executing unit tests (particularly in a containerized environment).  This
+# script installs the Pbench requirements; however, the expectation is that all
+# requirements will have been satisfied by previously-installed RPMs, so this
+# step serves mostly to verify that assertion.  This script also invokes the
+# project setup.py file which creates and installs the `pbench` Python-based
+# command scripts.  The two installations are performed with the `--user`
+# switch which ensures that their results land where Python will look for them
+# implicitly.  Therefore, it revokes any existing definition of PYTHONPATH, to
+# prevent other pbench installations from being found inadvertently.  Finally,
+# this script adds the directory containing the locally-installed pbench
+# commands to the front of the path (actually, this is done first to quiet
+# warnings from the installations) to ensure that the current versions are
+# chosen over any installed ones, and it adds `/usr/sbin` to the end of the
+# path for the `ip` command, used by pbench-register-tool.
 
-pip3 install --no-cache-dir --progress-bar off --no-color --prefix="${_prefix}" -r ${progdir}/lint-requirements.txt -r ${progdir}/docs/requirements.txt -r ${progdir}/agent/requirements.txt -r ${progdir}/server/requirements.txt -r ${progdir}/agent/test-requirements.txt -r ${progdir}/server/test-requirements.txt
-
-_pdir=${_prefix}/bin
-if [[ ":${PATH:-}:" != *":${_pdir}:"* ]]; then
-    export PATH=${_pdir}${PATH:+:${PATH}}
-fi
-
-NL=$'\n'
-while read -r _pdir; do
-    if [[ ":${PYTHONPATH:-}:" != *":${_pdir}:"* ]]; then
-        PYTHONPATH=${_pdir}${PYTHONPATH:+:${PYTHONPATH}}
-    fi
-done <<< "$(ls -1d ${_prefix}/lib*/python3.*/site-packages 2> /dev/null)${NL}$(head -n 1 ${_prefix}/lib*/python3.*/site-packages/pbench.egg-link 2> /dev/null)"
-export PYTHONPATH
-
-python3 setup.py develop --prefix="${_prefix}"
+PATH=${HOME}/.local/bin/:${PATH}:/usr/sbin
```

Comment:
While I suppose this is unlikely to change, I always "played safe" and used `$(python3 -m site --user-base)/bin` rather than assuming `~/.local/bin`.

Reply:
Yes -- I saw that in one of your PRs and meant to incorporate it here.  Will do.